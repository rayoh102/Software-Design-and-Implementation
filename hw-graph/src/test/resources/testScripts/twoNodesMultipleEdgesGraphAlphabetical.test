# Same as twoNodesMultipleEdgesGraph.test, but adding nodes in reverse order and adding edges
# in non-alphabetical order to test if it prints out in alphabetical order

# Create a graph
CreateGraph twoNodesMultipleEdgesGraphAlphabetical

# Add two nodes in reverse order
AddNode two_nodes_multiple_edges_graph_alphabetical b
AddNode two_nodes_multiple_edges_graph_alphabetical a

# Add five edges in non-alphabetical order
AddEdge twoNodesMultipleEdgesGraphAlphabetical a b AB1
AddEdge twoNodesMultipleEdgesGraphAlphabetical b b BB
AddEdge twoNodesMultipleEdgesGraphAlphabetical a b AB2
AddEdge twoNodesMultipleEdgesGraphAlphabetical b a BA
AddEdge twoNodesMultipleEdgesGraphAlphabetical a a AA

# Print the nodes in the graph, the outgoing edges from a and the outgoing edges from b
ListNodes twoNodesMultipleEdgesGraphAlphabetical
ListChildren twoNodesMultipleEdgesGraphAlphabetical a
ListChildren twoNodesMultipleEdgesGraphAlphabetical b