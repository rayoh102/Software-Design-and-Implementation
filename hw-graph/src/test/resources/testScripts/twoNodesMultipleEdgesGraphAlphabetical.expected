# Same as twoNodesMultipleEdgesGraph.test, but adding nodes in reverse order and adding edges
# in non-alphabetical order to test if it prints out in alphabetical order

# Create a graph
created graph twoNodesMultipleEdgesGraphAlphabetical

# Add two nodes in reverse order
added node b to twoNodesMultipleEdgesGraphAlphabetical
added node a to twoNodesMultipleEdgesGraphAlphabetical

# Add five edges in non-alphabetical order
added edge AB1 from a to b in twoNodesMultipleEdgesGraphAlphabetical
added edge BB from b to b in twoNodesMultipleEdgesGraphAlphabetical
added edge AB2 from a to b in twoNodesMultipleEdgesGraphAlphabetical
added edge BA from b to a in twoNodesMultipleEdgesGraphAlphabetical
added edge AA from a to a in twoNodesMultipleEdgesGraphAlphabetical

# Print the nodes in the graph, the outgoing edges from a and the outgoing edges from b
twoNodesMultipleEdgesGraphAlphabetical contains: a b
the children of a in twoNodesMultipleEdgesGraphAlphabetical are: a(AA) b(AB1) b(AB2)
the children of b in twoNodesMultipleEdgesGraphAlphabetical are: a(BA) b(BB)